src/lib/services/proposalService.ts

```diff
@@ -159,3 +165,77 @@ export const useProposalsCountByWalletAddress = (
     }
   );
 };
+
+export const useProposalList = (
+  offset: number,
+  pageSize: number,
+  statuses: ProposalStatus[],
+  types: ProposalType[],
+  search: string,
+  isMyProposal: boolean
+): UseQueryResult<Proposal[]> => {
+  const { indexerGraphClient } = useCelatoneApp();
+  const expression = useProposalListExpression(
```

Comment:
should we memoized this? @evilpeach

Reply:
You can use `useMemo` after converting this to a utility function.